src/lib/location.ts

```diff
@@ -107,80 +108,87 @@ export async function resolveLocation(
     };
   } else {
     const { address } = zipOrAddress;
-    const response = await geocoder.geocode(address, ['census2024']);
-    const result = response?.results?.at(0);
-
-    // Accuracy type "state" is too imprecise
-    if (
-      !result ||
-      result.address_components.country !== 'US' ||
-      result.accuracy_type === 'state'
-    ) {
-      return null;
-    }
-
-    const censusInfo = result.fields.census['2024'];
-
-    const geographies: Geography[] = [];
-
-    const state = await db.get<DbGeography>(
-      `SELECT * FROM geographies
-      WHERE type = 'state' AND state = ?`,
-      result.address_components.state,
-    );
-    if (state) {
-      geographies.push({ ...state, intersection_proportion: 1.0 });
-    }
-
-    const county = await db.get<DbGeography>(
-      `SELECT * FROM geographies
-      WHERE type = 'county' AND county_fips = ?`,
-      censusInfo.county_fips,
-    );
-    if (county) {
-      geographies.push({ ...county, intersection_proportion: 1.0 });
-    }
-
-    const point = turf.point([result.location.lng, result.location.lat]);
-
-    // There is a relatively small number of 'custom' geographies, so running
-    // the "contains" computation over all of them is quick. If necessary, we
-    // can speed this up by using a spatial index and/or caching the parsed
-    // GeoJSON in memory.
-    (
-      await db.all<DbGeography[]>(
-        `SELECT * FROM geographies WHERE type = 'custom'`,
+    try {
+      const response = await geocoder.geocode(address, ['census2024']);
```

Comment:
This is the only line that needs to be inside the try block; this line is what's throwing the exception. With everything else in the try block, any other exceptions that get thrown will get treated as a generic failure to resolve the location, which may end up hiding errors we would rather know about.